01-react-hooks/hooks-example/src/useCustom/class.jsx

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable react/prop-types */
+import React, {Component} from 'react';
+
+const searchForChuckJokes = async query => {
+  try {
+    const result = await fetch(`https://api.chucknorris.io/jokes/search?query=${query}`);
+    const resultJSON = await result.json();
+    return resultJSON.result || [];
+  } catch (error) {
+    console.error(error);
+    return [];
+  }
+};
+
+export class JokeList extends Component {
+  state = {
+    searching: false,
+    jokes: null
+  }
+
+  componentDidMount() {
+    this._handleNewSearch();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.query !== this.props.query) {
+      this._handleNewSearch();
+    }
+  }
+
+  _handleNewSearch = async() => {
```

Comment:
could have been `async _handleNewSearch() {`I guess

Reply:
if it's not an event handler, it might have been called `_seach`.. but the lesson is over, no need to update any more IMHO

Reply:
These are really minor notes. I should have checked this sooner.

Reply:
Well, yes, it could have been just `async _search {...}`, but I think this is more about individual code style preferences, I still like more the arrow function 🤷‍♂️